// gatsby-plugin-yandex-metrica/gatsby-ssr.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

var _react = _interopRequireDefault(require("react"));

var knownOptions = {
  clickmap: "boolean",
  trackLinks: "boolean",
  accurateTrackBounce: "boolean",
  webvisor: "boolean",
  trackHash: "boolean"
};

exports.onRenderBody = function (_ref, pluginOptions) {
  var setPostBodyComponents = _ref.setPostBodyComponents;

  if (process.env.NODE_ENV === "production") {
    var ymCreateOptions = {};

    for (var option in knownOptions) {
      if (typeof pluginOptions[option] === knownOptions[option]) {
        ymCreateOptions[option] = pluginOptions[option];
      }
    }

    return setPostBodyComponents([_react.default.createElement("script", {
      key: "gatsby-plugin-yandex-metrica",
      dangerouslySetInnerHTML: {
        __html: "(function(m,e,t,r,i,k,a){m[i]=m[i]||function(){(m[i].a=m[i].a||[]).push(arguments)};m[i].l=1*new Date();k=e.createElement(t),a=e.getElementsByTagName(t)[0],k.async=1,k.src=r,a.parentNode.insertBefore(k,a)})(window,document,\"script\",\"https://mc.yandex.ru/metrika/tag.js\",\"ym\");ym(" + pluginOptions.trackingId + ",\"init\"," + JSON.stringify(ymCreateOptions) + ");"
      }
    }), _react.default.createElement("noscript", null, _react.default.createElement("div", null, _react.default.createElement("img", {
      src: "https://mc.yandex.ru/watch/" + pluginOptions.trackingId,
      style: {
        position: "absolute",
        left: "-9999px"
      },
      alt: ""
    })))]);
  }

  return null;
};
